### PR TITLE
Change election course to elective course

### PR DESCRIPTION
**Pull request by Matthbo**
* Change election course names to elective course
* Description is now always a string, even if none exists

### CODE REVIEW

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java [hunk 1]
```diff
@@ -15,193 +15,192 @@
...
-                ElectionCourseDescription oldData = ElectionCourseStatements.getElectionCourseDescription(coursecode);
-                ElectionCourseDescription newData = new ElectionCourseDescription();
+            if (coursecode.equals(electiveCourseDescription.getCourseCode())) {
+                ElectiveCourseDescription oldData = ElectiveCourseStatements.getElectiveCourseDescription(coursecode);
+                ElectiveCourseDescription newData = new ElectiveCourseDescription();
                 newData.setCourseCode(oldData.getCourseCode());
-                if (!electionCourseDescription.getDescription().equals(oldData.getDescription())) {
-                    newData.setDescription(electionCourseDescription.getDescription());
+                if (!electiveCourseDescription.getDescription().equals(oldData.getDescription())) {
+                    newData.setDescription(electiveCourseDescription.getDescription());
```
[the remark this answers is not in the document]
Comment (guanhaowu) [reply Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java]: fixed

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java [hunk 2]
```diff
@@ -111,7 +111,7 @@ private ResponseEntity getAllElectiveCourseDescription() {
      */
     @ApiOperation(value = "Update a specific Elective Course for its description")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Successfully updated course description", response = ElectiveCourseDescription.class),
+            @ApiResponse(code = 200, message = "Successfully updated course description", response = ElectiveCourseDescription.class, responseContainer = "List"),
```
Comment (Bramgus12) [inline Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java]: You update a specific description of a elective course, so it is not possible to retrieve a list, yet you say that you return a list.

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java [hunk 3]
```diff
@@ -181,7 +181,7 @@ private ResponseEntity deleteElectiveCourseDescription(
     private ResponseEntity updateFile(@RequestParam MultipartFile file) {
         try {
             ElectiveCourseStatements.updateFile(file);
-            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+            return ResponseEntity.status(HttpStatus.OK).build();
```
Comment (Bramgus12) [inline Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java]: HTTP status code needs a body. You had the right status code but in the documentation it was not mentioned

File: Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java [hunk 4]
```diff
@@ -194,14 +194,14 @@ private ResponseEntity updateFile(@RequestParam MultipartFile file) {
...
-            @ApiResponse(code = 201, message = "Successfully uploaded"),
+            @ApiResponse(code = 200, message = "Successfully uploaded", response = ElectiveCourse.class),
             @ApiResponse(code = 400, message = "Bad request")
     })
     @PostMapping("/upload")
     private ResponseEntity uploadFile(@RequestParam("file") MultipartFile file) {
         try {
             ElectiveCourseStatements.uploadFile(file);
-            return ResponseEntity.status(HttpStatus.CREATED).build();
+            return ResponseEntity.status(HttpStatus.OK).build();
```
Comment (Bramgus12) [inline Back-end/src/main/java/com/bramgussekloo/projects/controller/ElectiveCourseController.java]: You say that there is a response but there isn't.

File: Back-end/src/main/java/com/bramgussekloo/projects/statements/ElectiveCourseStatements.java [hunk 5]
```diff
@@ -28,28 +31,45 @@
...
 
-    public static void uploadFile(MultipartFile file) throws IOException {
+    public static List<ElectiveCourse> uploadFile(MultipartFile file) throws IOException {
         File f = GetPropertyValues.getResourcePath("ElectiveCourse", fileNameVar);
+        FileNameMap fileNameMap = URLConnection.getFileNameMap();
+        System.out.println(fileNameMap);
+        String mimeType = fileNameMap.getContentTypeFor(file.getName());
+        System.out.println(mimeType);
+
+        assert (mimeType).equals("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
```
Comment (Bramgus12) [inline Back-end/src/main/java/com/bramgussekloo/projects/statements/ElectiveCourseStatements.java]: This is not bad i think. But maybe look at this way to get a mimeType. This is what I did to look up the mimeType of a .json file.
``` java
File f = GetPropertyValues.getResourcePath("Locations", file.getOriginalFilename());
String mimeType = Files.probeContentType(f.toPath());
```

**Comment by Bramgus12** [comment]
Not relevant anymore. Merged into API-improvements branch

**Pull request closed without merging**